andela/merry-ah-backend
# #162727462 signup and send verify link

#### What does this PR do?
Implements signup and send verify link to sign up users

#### Description of Task to be completed?
Users can sign up to Art Cave application providing at least the required fields.
Users get a verify email link to the email used for the registration

#### How should this be manually tested?
- Run `npm i`
- Create two databases for test and development using `createdb name`
- Copy the content of `.env.example` into a `.env` file and input your configuration details
- Run `sequelize db:migrate` to generate the models and tables
- start the server at `http://localhost:9000` using `npm run start:dev`
- Make a POST request to `http://localhost:9000/api/v1/auth/signup` providing the required fields for sign up.
- On success check your email inbox for the verify link
- Run `npm test`

#### Any background context you want to provide?
The required fields are
- firstname
- lastname
- username
- email
- password
- bio
- imgURL
- userType

#### What are the relevant pivotal tracker stories?
[#162727462](https://www.pivotaltracker.com/story/show/163052933)

#### Screenshots (if appropriate)
N/A
#### Questions:
N/A
